services/DataService.py

```diff
@@ -1,46 +1,31 @@
 from datetime import datetime
-from sqlalchemy.sql.sqltypes import String
+
 from controllers import db
-from models.AccountModel import Account
-from models.DataModel import Data
-from models.PredictionModel import Prediction
-from models.QAModel import QAModel
+from models import Data, QAModel
 
 
 class DataService():
-    '''Service class to handle all database CRUD operations for Data model'''
-
-    db_conn = db
-
-    def create_data(self, datum:Data)->Data:
-        '''Service function to create a new model in the database'''
-
-        self.db_conn.session.add(datum)
-        self.db_conn.session.commit()
+    
+    def create_data(self, datum:Data) -> Data:
+        db.session.add(datum)
+        db.session.commit()
         saved_data = self.read_data_by_id(datum.id)
         return saved_data
 
-    def read_data_by_id(self, id: int)->Data:
-        '''Service function to read a datum model from the database by id'''
+    def read_data_by_id(self, id: str) -> Data:
         selected_data = Data.query.filter(Data.id == id).first()
         return selected_data
 
-    def read_all_data_since(self, date:datetime)->list:
-        '''Service function to read all data models created since a given datetime'''
-
+    def read_all_data_since(self, date: datetime) -> list:
         selected_datas = Data.query.filter(Data.created_date > date).all()
         return selected_datas
 
-    def read_last_x_datum(self, x:int)->list:
-        '''Service function to read x number of most recently created data models'''
-
-        selected_datas = Data.query.order_by(Data.datum_id.desc()).limit(x)
+    def read_last_x_datum(self, x: int) -> list:
+        selected_datas = Data.query.order_by(Data.id.desc()).limit(x)
         selected_datas = selected_datas [::-1]
         return selected_datas
```

Comment:
Not sure what selected_data[::-1] is doing here, shouldn't we just return selected datas as it is retrieved from the Data query?

Reply:
I think all this does is reverse the list, so it's in ascending order by id

Reply:
Ahh, okay